go/oasis-test-runner/oasis/validator.go

```diff
@@ -81,26 +84,41 @@ func (val *Validator) startNode() error {
 func (net *Network) NewValidator(cfg *ValidatorCfg) (*Validator, error) {
 	valName := fmt.Sprintf("validator-%d", len(net.validators))
 
-	val := &Validator{
-		net:           net,
-		entity:        cfg.Entity,
-		consensusPort: net.nextNodePort,
-		grpcDebugPort: net.nextNodePort + 1,
+	valDir, err := net.baseDir.NewSubDir(valName)
+	if err != nil {
+		net.logger.Error("failed to create validator subdir",
+			"err", err,
+			"validator_name", valName,
+		)
+		return nil, errors.Wrap(err, "oasis/validator: failed to create validator subdir")
 	}
 
-	var err error
-	if val.dir, err = net.baseDir.NewSubDir(valName); err != nil {
-		net.logger.Error("failed to create valdiator subdir",
+	// Pre-provision the node identity, so that we can pass the validator's
+	// consensus public key to the node registration command.
+	signerFactory := fileSigner.NewFactory(valDir.String(), signature.SignerNode, signature.SignerP2P, signature.SignerConsensus)
+	valIdentity, err := identity.LoadOrGenerate(valDir.String(), signerFactory)
+	if err != nil {
+		net.logger.Error("failed to provision validator identity",
 			"err", err,
 			"validator_name", valName,
 		)
-		return nil, errors.Wrap(err, "oasis/validator: failed to create validator subdir")
+		return nil, errors.Wrap(err, "oasis/validator: failed to provision validator identity")
+	}
+
+	valConsensusPublicKey := valIdentity.ConsensusSigner.Public()
+
+	val := &Validator{
+		net:           net,
+		dir:           valDir,
+		entity:        cfg.Entity,
+		consensusPort: net.nextNodePort,
+		grpcDebugPort: net.nextNodePort + 1,
 	}
 
 	args := []string{
 		"registry", "node", "init",
 		"--datadir", val.dir.String(),
-		"--node.consensus_address", fmt.Sprintf("127.0.0.1:%d", val.consensusPort),
+		"--node.consensus_address", fmt.Sprintf("%s@127.0.0.1:%d", valConsensusPublicKey, val.consensusPort),
```

Comment:
Shouldn't this use the new consensus address structure to do the formatting?

Reply:
Yes, good idea.

I've implemented that.